test/unittests/evm_storage_test.cpp

```diff
@@ -169,6 +170,120 @@ TEST_P(evm, sstore_cost)
     }
 }
 
+TEST_P(evm, sstore_cost_legacy)
+{
+    static constexpr auto O = 0x000000000000000000_bytes32;
+    static constexpr auto X = 0xfeffffffffffffffff_bytes32;
+    static constexpr auto Y = 0x011000000000000002_bytes32;
+    static constexpr auto Z = 0x800000000000000001_bytes32;
+    static constexpr auto key = 0xef_bytes32;
+
+    static constexpr int64_t b = 9;  // Cost of other instructions.
+
+    static constexpr struct
+    {
+        int64_t set = 20000;
+        int64_t reset = 5000;
+        int64_t clear = 15000;
+    } c;
+
+    auto test = [this](const evmc::bytes32& original, const evmc::bytes32& current,
+                    const evmc::bytes32& value, int64_t expected_gas_used,
+                    int64_t expected_gas_refund) {
+        auto& storage = host.accounts[msg.recipient].storage;
+        storage[key] = {current, original};
+        execute(sstore(0xef, calldataload(0)), value);
```

Comment:
maybe could be simpler
```suggestion
        execute(sstore(0xef, value), {});
```